src/web/core/src/main/java/org/geoserver/web/LoginFormHTMLInclude.java

```diff
@@ -48,17 +56,23 @@ public void onComponentTagBody(final MarkupStream markupStream, final ComponentT
     @Override
     protected String importAsString() {
         try {
-            InputStream inputStream = this.resourceReference.getResource().getResourceStream()
-                    .getInputStream();
-            BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
-            StringBuilder sb = new StringBuilder();
-            String line;
-            while ((line = br.readLine()) != null) {
-                sb.append(line + "\n");
+
+            Configuration cfg = new Configuration( );
```

Comment:
Creating the configuration and loading the templates is expensive, you should consider following up this patter (without copying the KML comment maybe :-p ):
https://github.com/geoserver/geoserver/blob/master/src/wms/src/main/java/org/geoserver/wms/featureinfo/HTMLFeatureInfoOutputFormat.java#L51